core/vm/message/message.hpp

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef CPP_FILECOIN_CORE_VM_MESSAGE_HPP
+#define CPP_FILECOIN_CORE_VM_MESSAGE_HPP
+
+#include <boost/variant.hpp>
+#include <gsl/span>
+
+#include "common/outcome.hpp"
+#include "crypto/bls/bls_provider.hpp"
+#include "crypto/secp256k1/secp256k1_provider.hpp"
+#include "primitives/address/address.hpp"
+#include "primitives/big_int.hpp"
+
+namespace fc::vm::message {
+
+  /**
+   * @brief Message error codes
+   */
+  enum class MessageError {
+    INVALID_LENGTH = 1,
+    INVALID_KEY_LENGTH,
+    INVALID_SIGNATURE_LENGTH,
+    WRONG_SIGNATURE_TYPE
+  };
+
+  using primitives::BigInt;
+  using primitives::address::Address;
+
+  /**
+   * @brief UnsignedMessage struct
+   */
+  struct UnsignedMessage {
+    Address to;
+    Address from;
+
+    uint64_t nonce;
+
+    BigInt value;
+
+    uint64_t method;
```

Comment:
The order may be important to not get confused in case of serialization (I'm not sure if it is required).
Originally, the order is {to, from, nonce, value, gas price, gas limit, method, params}.

Reply:
agree. The order of field members follows the spec while the order in serialized form matches Lotus implementation (which is a must for compatibility purpose)